quickannotator/api/v1/tile/helper.py

```diff
@@ -76,35 +79,62 @@ def generate_random_circle_within_bbox(bbox: Polygon, radius: float) -> shapely.
     intersection = bbox.intersection(circle)
     return intersection
 
-def compute_on_tile(db, qadb, tile_id: int, sleep_time=5): 
-    def async_task():
-        Session = sessionmaker(bind=db.engine)
-
-        time.sleep(sleep_time)
-
+@ray.remote
+def remote_compute_on_tile(db_url, tile_id: int, sleep_time=5):
+    time.sleep(sleep_time)
+    # Create the engine and session for each Ray task
+    engine = create_engine(db_url)
+    Session = sessionmaker(bind=engine)
+
+    # Attach the event listener to the engine (inside the task to ensure it's unique per task)
+    @event.listens_for(engine, "connect")
+    def connect(dbapi_connection, connection_record):
+        try:
+            # Enable the Spatialite extension on the raw SQLite connection
+            if hasattr(dbapi_connection, "enable_load_extension"):
+                dbapi_connection.enable_load_extension(True)
+                dbapi_connection.execute('SELECT load_extension("mod_spatialite")')
+                dbapi_connection.execute("PRAGMA busy_timeout=10000000;")
+                print("Spatialite extension loaded successfully")
+            else:
+                print("Extension loading not supported for this connection")
+        except Exception as e:
+            print(f"Error enabling Spatialite extension: {e}")
+
+    try:
+        # Start a session for the task
         with Session() as session:
-            tile = tile_by_id(session, tile_id)
+            # Example: load the tile and process
+            tile = tile_by_id(session, tile_id)  # Replace with your actual function to get the tile
             image_id: int = tile.image_id
             annotation_class_id: int = tile.annotation_class_id
 
+            # Process the tile (using shapely for example)
             bbox = shapely.wkb.loads(tile.geom.data)
-            for i in range(random.randint(20, 40)):
+            for _ in range(random.randint(20, 40)):
                 polygon = generate_random_circle_within_bbox(bbox, 100)
                 insert_new_annotation(session, image_id, annotation_class_id, is_gt=False, polygon=polygon)
 
-            # now that the tile has been processed, update the tile.seen column to 2
+            # Mark tile as processed
             tile.seen = 2
```

Comment:
normally you want to define these as constants to make the code a bit easier to read for 3rd parties. i.e., tile.seen = qa.constants.SEEN   (or something like that)